Python/4-cache.py

```diff
@@ -0,0 +1,26 @@
+def fn():
+    print('Generate cache')
+    cache = {}
+    def get_from_cache(key):
+        res = cache.get(key)
+        if res:
```

Comment:
An explicit `if res is not None` would be more reliable here; `if res` will fail if `res` is `0`, `False` or `''`.